api/src/wfl/sink.clj

```diff
@@ -242,21 +243,104 @@
                                 :fromOutputs fromOutputs}))))
     (assoc request :dataset dataset')))
 
+(defn ^:private push-job [{:keys [details] :as _sink} job]
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (jdbc/insert! tx details job)))
+
+(defn ^:private peek-job-queue [{:keys [details] :as _sink}]
+  (let [query "SELECT * FROM %s
+               WHERE status = 'succeeded' AND consumed IS NULL
+               ORDER BY id LIMIT 1"]
+    (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+      (postgres/throw-unless-table-exists tx details)
+      (first (jdbc/query tx (format query details))))))
+
+(defn ^:private pop-job-queue!
+  [{:keys [details] :as _sink} {:keys [id] :as _job}]
+  {:pre [(some? id) (integer? id)]}
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (jdbc/update! tx details {:consumed (utc-now)} ["id = ?" id])))
+
+(defn ^:private update-datarepo-job-statuses
+  "Fetch and record the status of all 'running' jobs from tdr created by the
+   `sink`. Throws `UserException` when any new job status is `failed`."
+  [{:keys [details] :as _sink}]
+  (letfn [(read-active-jobs []
+            (let [query "SELECT id, job, workflow FROM %s
+                         WHERE status = 'running'
+                         ORDER BY id ASC"]
+              (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+                (map (juxt :id :job :workflow)
+                     (jdbc/query tx (format query details))))))
+          (write-job-status [job-id status]
```

Comment:
The `job-id` name here is confusing.  Is this meant to be the row ID or the job ID, from where this is called / how this is used I think row ID?

Reply:
It's meant to be the job ID. Its called on line 283 within this function.